### PR TITLE
Simplified proxy, item cache related stuff removed

### PR DESCRIPTION
**Pull request by annando**
With the new handling of attachments and contact avatars we only use the proxy functionality for some rare cases. So we can simplify the functionality. We don't store proxy files anymore in the system, so we don't need the item cache stuff anymore as well.

### CODE REVIEW

**Comment by AlfredSK** [comment]
Is the /proxy folder still used? If it isn't used anymore, is the content cleaned up automatically? Otherwise, can it be safely deleted?

**Comment by annando** [comment]
The folder can be removed. It is not cleaned automatically since then we would had to keep most of the removed functionality.

**Comment by AlfredSK** [comment]
> The folder can be removed.

Oh! I'm looking forward to free up 310 GB storage space.

**Comment by annando** [comment]
**310 GB** in that single folder? On squeet.me it's 1.9 GB.

**Comment by AlfredSK** [comment]
Yes. Is that unusual? :-)

```
root@librabox2 ~ # du -sh html/mordor/proxy/                                                                                                                       
3,8G    html/mordor/proxy/
root@librabox2 ~ # du -sh html/venera/proxy/
70G     html/venera/proxy/
root@librabox2 ~ # du -sh html/libranet/proxy/
236G    html/libranet/proxy/
```

mordor.social is more or less a single user node - one really active user. And even that tiny node has 3.8 GB in /proxy.

**Comment by annando** [comment]
Sounds like the cleaning job failed. Just another argument for the drastic redesign.

**Comment by tobiasd** [comment]
Could you add a note about this to the CHANGELOG file. Something like

> Enhanced the image proxy, the /proxy directory can be removed [annando]

File: include/api.php [hunk 1]
```diff
@@ -2654,7 +2653,7 @@ function api_add_attachments_to_body(array $item)
  * @return array
  * @throws InternalServerErrorException
  */
-function api_get_attachments(&$body)
+function api_get_attachments(&$body, $uriid)
```
Comment (tobiasd) [inline include/api.php:2657]: Can you document that new parameter?

Comment (annando) [reply include/api.php:2657]: Done

**Comment by annando** [comment]
I guess this PR is now complete.

**Comment by AlfredSK** [comment]
Tests are failing.

**Comment by annando** [comment]
@MrPetovan sorry for adding more and more stuff to this PR. But I guess that now this is really finished.

**Review by MrPetovan — APPROVED** [review]
I can't review that many files at once, I trust you will fix the bits that will be failing over the next weeks?

**Comment by annando** [comment]
@MrPetovan I hope you had a short glimpse at the new structure (new table, new class, ...)? Over the next days I guess I will add more functionality so that we really use the new `uri-id` fields in the contact tables.

**Comment by MrPetovan** [comment]
I've been wondering about this new foreign key in the contact table, but I didn't have it in me to wade through 58 files to find the purpose.

**Comment by annando** [comment]
The `contact`, `apcontact` and `fcontact` are now linked to the `uri-id` table. Before this we had to connect them via the `url` and `nurl` fields.

**Comment by MrPetovan** [comment]
That's what I was expecting, but what kind of improvement will it bring to us?

**Comment by annando** [comment]
With this field we can now create (for example) a view that connects the public contact (uid=0 in the contact table), the personal contact (uid != 0 in the contact table), the AP related information (in apcontact) and the Diaspora related information (in fcontact). In the future we then can switch to this view (like we had done in the "post" handling). And in some following step we can then replace the underlying tables without any problem.

**Pull request merged**